packages/blade/src/components/Switch/_decisions/decisions.md

```diff
@@ -0,0 +1,85 @@
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=31366-831710&t=qzTLJeIDluL4Ps23-0) to all variants of the Switch component
+
+
+## `Switch` API
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| isChecked | `boolean` | `false` | State of the switch, if explicitly set the switch will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components) |        |
+| defaultChecked | `boolean` | `false` | Default state of the switch, if isChecked is not provided the switch will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) |        |
+| onChange | `({ event, isChecked, value }) => void` | `undefined` | The function to be called when the switch state changes. |        |
+| isDisabled | `boolean` | `false` | Control whether the switch is disabled or not. |        |
+| accessibilityLabel | `string` | `undefined` | Provides accessible label for internal checkbox component that sets the aria-label prop for screen readers. |        |
+| size | `small, medium` | `medium` | Size of the switch component |        |
+| name (html) | `string` | `undefined` | The name of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |        |
+| value (html) | `string` | `undefined` | The value of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |        |
+| id (html) | `string` | `undefined` | The id of the input field in a switch, useful for associating a label element with the input via `htmlFor` prop |        |
+
+### Examples:
+
+#### Basic
+
+```js
+// basic
+<Switch value="dark-mode" />
+```
+
+#### Controlled, Uncontrolled
+
+```js
+const Controlled = () => {
+  const [chceked, setChecked] = React.useState(false);
+
+  return (
+    <Switch isChecked={checked} onChange={({isChecked}) => setChecked(isChecked)} value="dark-mode" />
+  );
+};
+
+const Uncontrolled = () => {
+  return (
+    <Switch defaultChecked={true} value="dark-mode" />
+  );
+};
+```
+
+#### Custom Label
+
+Switch by default doesn't render any `<label>` element internally, if users want they can provide it: 
+
+```js
+<Box display="flex">
+  <label htmlFor="dark-mode">
+    Dark mode
+  </label>
+  <Switch id="dark-mode" />
+</Box>
+
+// or
+
+<label>Dark mode <Switch /></label>
+```
+
+## Accessibility
+
+Switch will work similarly as Checkbox, where the hidden input field will follow the [Checkbox accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/) pattern. But we will also have a visual button which will follow the [Switch accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/switch/) pattern. 
```

Comment:
didn't quite understand this statement? 🤔

Reply:
Sorry for the confusion, updated with better wording and example from WAI ARIA site.